cmd/kubectl-gadget/utils/trace.go

```diff
@@ -507,23 +469,23 @@ func waitForCondition(traceID string, conditionFunction func(*gadgetv1alpha1.Tra
 	// We only watch the traces if there are some which did not already satisfy
 	// the conditionFunction.
 	if len(satisfiedTraces)+len(erroredTraces) < tracesNumber {
-		var watcher watch.Interface
-
-		// We will need to watch events on them.
-		// For this, we will get a watcher on all the traces which share the same
-		// ID.
-		// Get a watcher on all the traces which have the same ID.
-		// Indeed, all the traces on different nodes but linked to one gadget share
+		// We will need to list and watch events on them.
+		// For this, we will get a ListerWatcher on all the traces which share the
+		// same ID.
+		// NOTE all the traces on different nodes but linked to one gadget share
 		// the same ID.
 		// We will also begin to monitor events since the above GET of the traces
-		// list.
-		watcher, err = getTraceWatcher(traceID, traceList.ListMeta.ResourceVersion)
+		// list thanks to resource version.
+		traceListerWatcher, err := getTraceListerWatcher(traceID, traceList.Items[tracesNumber-1].ResourceVersion)
```

Comment:
What is the difference between `traceList.ListMeta.ResourceVersion` and `traceList.Items[tracesNumber-1].ResourceVersion`?